### PR TITLE
Update source-declarative-manifest to CDK version 6.X

### PR DESCRIPTION
**Pull request by maxi297**
## What
Update source-declarative-manifest to CDK version 6.X

## How
Following [the migration guide](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/cdk-migrations.md#upgrading-to-6xx), we need to pass the catalog, config and state when initializing the source now.

## User Impact
Example using source-datascope

### Output on version 0.2.3
```
{"type":"CATALOG","catalog":{"streams":[{"name":"locations","json_schema":{"$schema":"http://json-schema.org/draft-04/schema#","type":"object","properties":{"id":{"type":"integer"},"name":{"type":"string"},"description":{"type":["string","null"]},"code":{"type":["string","null"]},"address":{"type":["string","null"]},"city":{"type":["string","null"]},"country":{"type":["string","null"]},"latitude":{"type":["number","null"]},"longitude":{"type":["number","null"]},"region":{"type":["string","null"]},"phone":{"type":["string","null"]},"company_code":{"type":["string","null"]},"company_name":{"type":["string","null"]}}},"supported_sync_modes":["full_refresh"],"source_defined_primary_key":[["id"]],"is_resumable":true},{"name":"answers","json_schema":{"$schema":"http://json-schema.org/draft-07/schema#","type":"object","properties":{"form_name":{"type":"string"},"form_state":{"type":["string","null"]},"user_name":{"type":"string"},"user_identifier":{"type":"string"},"code":{"type":"string"},"form_id":{"type":"integer"},"created_at":{"type":"string"},"form_answer_id":{"type":"integer"},"latitude":{"type":["number","null"]},"longitude":{"type":["number","null"]},"[question_name1]":{"type":"string"},"[question_name2]":{"type":"string"},"[question_name3]":{"type":"string"}}},"supported_sync_modes":["full_refresh","incremental"],"source_defined_cursor":true,"default_cursor_field":["created_at"],"source_defined_primary_key":[["form_answer_id"]],"is_resumable":true},{"name":"lists","json_schema":{"$schema":"http://json-schema.org/draft-07/schema#","type":"object","properties":{"id":{"type":"integer"},"name":{"type":"string"},"description":{"type":"string"},"attribute1":{"type":"string"},"attribute2":{"type":"string"},"list_id":{"type":"integer"},"account_id":{"type":"integer"},"code":{"type":"string"},"created_at":{"type":"string"},"updated_at":{"type":"string"}}},"supported_sync_modes":["full_refresh"],"source_defined_primary_key":[["id"]],"is_resumable":true},{"name":"notifications","json_schema":{"$schema":"http://json-schema.org/draft-04/schema#","type":"object","properties":{"id":{"type":"integer"},"type":{"type":"string"},"url":{"type":"string"},"form_name":{"type":"string"},"form_code":{"type":"string"},"user":{"type":"string"},"created_at":{"type":"string"}}},"supported_sync_modes":["full_refresh","incremental"],"source_defined_cursor":true,"default_cursor_field":["created_at"],"source_defined_primary_key":[["id"]],"is_resumable":true}]}}
```

### Output on version 0.2.4
```
{"type":"CATALOG","catalog":{"streams":[]}}
```

### Output on fixed version
* Copying source-datascope manifest in `airbyte-integrations/connectors/source-declarative-manifest/source_declarative_manifest/manifest.yaml`
* Updating [this line](https://github.com/airbytehq/airbyte/blob/eb18c21d43eeec789f2cba16e10bbc94f795ac52/airbyte-integrations/connectors/source-declarative-manifest/source_declarative_manifest/run.py#L41) to reference the copied manifest

The output is the following:
```
{"type":"CATALOG","catalog":{"streams":[{"name":"answers","json_schema":{"$schema":"http://json-schema.org/draft-07/schema#","type":"object","properties":{"form_name":{"type":"string"},"form_state":{"type":["string","null"]},"user_name":{"type":"string"},"user_identifier":{"type":"string"},"code":{"type":"string"},"form_id":{"type":"integer"},"created_at":{"type":"string"},"form_answer_id":{"type":"integer"},"latitude":{"type":["number","null"]},"longitude":{"type":["number","null"]},"[question_name1]":{"type":"string"},"[question_name2]":{"type":"string"},"[question_name3]":{"type":"string"}}},"supported_sync_modes":["full_refresh","incremental"],"source_defined_cursor":true,"default_cursor_field":["created_at"],"source_defined_primary_key":[["form_answer_id"]],"is_resumable":true},{"name":"notifications","json_schema":{"$schema":"http://json-schema.org/draft-04/schema#","type":"object","properties":{"id":{"type":"integer"},"type":{"type":"string"},"url":{"type":"string"},"form_name":{"type":"string"},"form_code":{"type":"string"},"user":{"type":"string"},"created_at":{"type":"string"}}},"supported_sync_modes":["full_refresh","incremental"],"source_defined_cursor":true,"default_cursor_field":["created_at"],"source_defined_primary_key":[["id"]],"is_resumable":true},{"name":"locations","json_schema":{"$schema":"http://json-schema.org/draft-04/schema#","type":"object","properties":{"id":{"type":"integer"},"name":{"type":"string"},"description":{"type":["string","null"]},"code":{"type":["string","null"]},"address":{"type":["string","null"]},"city":{"type":["string","null"]},"country":{"type":["string","null"]},"latitude":{"type":["number","null"]},"longitude":{"type":["number","null"]},"region":{"type":["string","null"]},"phone":{"type":["string","null"]},"company_code":{"type":["string","null"]},"company_name":{"type":["string","null"]}}},"supported_sync_modes":["full_refresh"],"source_defined_primary_key":[["id"]],"is_resumable":true},{"name":"lists","json_schema":{"$schema":"http://json-schema.org/draft-07/schema#","type":"object","properties":{"id":{"type":"integer"},"name":{"type":"string"},"description":{"type":"string"},"attribute1":{"type":"string"},"attribute2":{"type":"string"},"list_id":{"type":"integer"},"account_id":{"type":"integer"},"code":{"type":"string"},"created_at":{"type":"string"},"updated_at":{"type":"string"}}},"supported_sync_modes":["full_refresh"],"source_defined_primary_key":[["id"]],"is_resumable":true}]}}
```

As you can see, the order of the streams changes but this should not have any impact.

## Can this PR be safely reverted and rolled back?
- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by bnchrch — APPROVED** [review]
LGTM

If we merge this, we can generate a dev release for source-datascope and source-coingecko to test that it fully works

File: airbyte-integrations/connectors/source-declarative-manifest/source_declarative_manifest/run.py [hunk 1]
```diff
@@ -48,8 +60,39 @@ def handle_command(args: List[str]) -> None:
         handle_remote_manifest_command(args)
 
 
+def _get_local_yaml_source(args: List[str]):
+    catalog_path = AirbyteEntrypoint.extract_catalog(args)
```
[the remark this answers is not in the document]
Comment (maxi297) [reply airbyte-integrations/connectors/source-declarative-manifest/source_declarative_manifest/run.py:64]: This is a good question. I guess it is a balance between allowing the source to have their own constructor interface and standardization. We can probably have that for low-code though

**Review by brianjlai — APPROVED** [review]
changes look good. pending the functional testing of the new SDM base image

**Comment by octavia-approvington** [comment]
Merged using the approve bot.

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**